### PR TITLE
Ct 488/migrate invalid model tests

### PR DESCRIPTION
**Pull request by VersusFacit**
Contributes to 488. 

### Description

Converts 011_invalid_data_tests. There were a few tests that had naming did that not reflect what was being test or was testing for the wrong errors functionally. I've corrected that where possible.

I also added a new fixture that allows for adding properties to model dirs. This proves useful for edge cases where you want to override properties or models without overriding the other.

### Checklist

- [X] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [X] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [X] I have run this code in development and it appears to resolve the stated issue
- [X] This PR includes tests, or tests are not required/relevant for this PR
- [X] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [X] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#Adding-CHANGELOG-Entry)

### CODE REVIEW

**Review by iknox-fa — CHANGES_REQUESTED** [review]
LGTM other than the changelog.

File: .changes/unreleased/Under the Hood-20220425-001112.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+kind: Under the Hood
```
[the remark this answers is not in the document]
Comment (iknox-fa) [reply .changes/unreleased/Under the Hood-20220425-001112.yaml:1]: Confirmed: https://dbt-labs.slack.com/archives/C0131TY7EEA/p1650900574658189

[the remark this answers is not in the document]
Comment (VersusFacit) [reply .changes/unreleased/Under the Hood-20220425-001112.yaml:1]: Oh great! That makes a lot more sense to me actually. I'll nix that.

**Review by iknox-fa — APPROVED** [review]
LGTM!
